### PR TITLE
Fix asset destroy cache bug

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

- When `GlTFResource` or `Entity` destroy the asset cache still residual.
This PR fix this and clear the dirty cache when `GlTFResource` or `Entity` destroy.

- Fix spelling errors.

### CODE REVIEW

File: packages/core/src/base/EngineObject.ts [hunk 1]
```diff
@@ -15,14 +15,35 @@ export abstract class EngineObject {
...
   }
+
+  /**
+   * Destroy self.
+   */
+  destroy(): void {
+    if (this._destroyed) return;
+
+    const resourceManager = this._engine.resourceManager;
+    resourceManager?._deleteAsset(this);
```
Comment (eyworldwide) [inline packages/core/src/base/EngineObject.ts:45]: Need not to create a variable used once:
```typescript
this._engine.resourceManager?._deleteAsset(this);
```

File: packages/loader/src/scene-loader/Oasis.ts [hunk 2]
```diff
@@ -15,9 +15,11 @@ export class Oasis extends EventDispatcher {
   private schema: Schema;
   public timeout: number;
   private oasis = this;
+  public engine:Engine;
 
   private constructor(private _options: Options, public readonly pluginManager: PluginManager) {
-    super(_options.engine);
+    super();
+    this.engine=_options.engine;
```
Comment (eyworldwide) [inline packages/loader/src/scene-loader/Oasis.ts:22]: format the code

File: packages/core/src/base/EngineObject.ts (hunk 1 above)
Comment (GuoLei1990) [reply packages/core/src/base/EngineObject.ts]: done

File: packages/loader/src/scene-loader/Oasis.ts (hunk 2 above)
Comment (GuoLei1990) [reply packages/loader/src/scene-loader/Oasis.ts]: done

**Review by eyworldwide — APPROVED** [review]
+1

**Pull request merged**